### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by olegsobchuk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,16 +41,128 @@ end
...
-don't use instance variables, instead use `let`
+[Better Specs](http://betterspecs.org/)
+
+***
+
+use [FactoryGirl](https://github.com/thoughtbot/factory_girl) for build instances
+
+***
+
+in FactoryGirl when you define attributes don't use `{}` if you use stable value
```
Comment (mushkin-v) [inline README.md]: if you use stable value when you define attributes in FactoryGirl don't use `{}`

File: README.md [hunk 2]
```diff
@@ -41,16 +41,128 @@ end
...
+FactoryGirl.define do
+  factory :article do
+    published true
+  end
+end
+```
+
+***
+
+use [Faker](https://github.com/stympy/faker) for generate rundom attributes
```
Comment (mushkin-v) [inline README.md:76]: use bla-bla-bla for generation of random attributes

File: README.md [hunk 3]
```diff
@@ -41,16 +41,128 @@ end
...
+
+***
+
+use [DatabaseCleaner](https://github.com/DatabaseCleaner/database_cleaner) for clean database arount tests
```
Comment (mushkin-v) [inline README.md]: use bla-bla-bla for cleaning database data which is related to tests

File: README.md [hunk 4]
```diff
@@ -41,16 +41,128 @@ end
...
 
 ```
-#good
+# good
 let(:user) { FactoryGirl.create(:user) }
 ```
+
+***
+
+when you wright tests use same order as methods defined
```
Comment (mushkin-v) [inline README.md]: when you write tests use the same order as defined methods

File: README.md [hunk 5]
```diff
@@ -41,16 +41,128 @@ end
...
+    User.skip_callback(:create, :before, :build_profile)
+  end
+end
+```
+
+if you need this callback in test just set it in `before` block and skip it in `after` block
+
+***
+
+don't save objects to database if don't need it
```
Comment (mushkin-v) [inline README.md]: don't save objects to database if you don't need them

File: README.md [hunk 6]
```diff
@@ -41,16 +41,128 @@ end
 
 ## TESTS
 
-don't use instance variables, instead use `let`
+[Better Specs](http://betterspecs.org/)
```
Comment (olegsobchuk) [inline README.md]: test

**Pull request merged**